src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp

```diff
@@ -0,0 +1,546 @@
+
+// #define DP_ROUND_RZ(a)   _mm_cvtsd_f64(_mm_roundscale_sd(_mm_setzero_pd(), _mm_set_sd(a), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC)))
+
+// #define DP_CONST(C)    _castu64_f64(0x##C##ull)
+// #define DP_AND(X, Y)   _mm_cvtsd_f64(_mm_and_pd(_mm_set_sd(X), _mm_set_sd(Y)))
+// #define DP_XOR(X, Y)   _mm_cvtsd_f64(_mm_xor_pd(_mm_set_sd(X), _mm_set_sd(Y)))
+// #define DP_OR(X, Y)    _mm_cvtsd_f64(_mm_or_pd(_mm_set_sd(X), _mm_set_sd(Y)))
+// #define DP_DIV_RZ(a, b) __fence(_mm_cvtsd_f64(_mm_div_round_sd(_mm_set_sd(a), _mm_set_sd(b), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC))))
+// #define DP_FNMA(a, b, c)    __fence(_mm_cvtsd_f64(_mm_fnmadd_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c))))
+// #define DP_FNMA_RZ(a, b, c) __fence(_mm_cvtsd_f64(_mm_fnmadd_round_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC))))
+
+// #define D2L(x)  _mm_castpd_si128(x)
+// // transfer highest 32 bits (of low 64b) to GPR
+// #define TRANSFER_HIGH_INT32(X)   _mm_extract_epi32(D2L(_mm_set_sd(X)), 1)
+
+
+// double fmod(double x, double y)
+// {
+  __ subptr(rsp, 0x28);
+  __ mov64(rax, 0x7fffffffffffffff);
+  __ movapd(xmm20, xmm0);
+  __ movapd(xmm3, xmm1);
+// double a, b, sgn_a, q, bs, bs2;
+// unsigned eq;
+
+//     // |x|, |y|
+//     a = DP_AND(x, DP_CONST(7fffffffffffffff));
+  __ vxorpd(xmm18, xmm18, xmm18, Assembler::AVX_512bit);
+  __ vmovsd(xmm5, xmm18, xmm20);
+  __ movq(xmm17, rax);
+  __ vandpd(xmm0, xmm5, xmm17, Assembler::AVX_512bit);
+//     b = DP_AND(y, DP_CONST(7fffffffffffffff));
+  __ vandpd(xmm1, xmm3, xmm17, Assembler::AVX_512bit);
+//   // sign(x)
+//   sgn_a = DP_XOR(x, a);
+  __ vmovsd(xmm17, xmm18, xmm0);
+  __ vxorpd(xmm16, xmm5, xmm17, Assembler::AVX_512bit);
+
+//   q = DP_DIV_RZ(a, b);
+  __ vmovsd(xmm5, xmm18, xmm1);
+  __ evdivsd(xmm2, xmm17, xmm5, Assembler::EVEX_RZ);
+//   q = DP_ROUND_RZ(q);
+  __ vmovsd(xmm4, xmm18, xmm2);
+  __ vrndscalesd(xmm19, xmm18, xmm4, 0x0B);
+
+//   eq = TRANSFER_HIGH_INT32(q);
+  __ vmovsd(xmm4, xmm18, xmm19);
+  __ pextrd(rax, xmm4, 1);
+
+//   if (!eq)  return x + sgn_a;
+__ testl(rax, rax);
+__ jcc(Assembler::equal, L_12ca);
+
+//   if (eq >= 0x7fefffffu) goto SPECIAL_FMOD;
+__ cmpl(rax, 0x7FEFFFFF);
+__ jcc(Assembler::below, L_1280);
+
+// SPECIAL_FMOD:
+
+//   // y==0 or x==Inf?
+//   if ((b == 0.0) || (!(a <= DP_CONST(7fefffffffffffff))))
+  __ vxorpd(xmm2, xmm2, xmm2, Assembler::AVX_512bit);
+  __ ucomisd(xmm1, xmm2);
+  __ jcc(Assembler::parity, L_115a);
+  __ jcc(Assembler::equal, L_1271);
+
+  __ bind(L_115a);
+  __ mov64(rax, 0x7FEFFFFFFFFFFFFF);
+  __ movq(Address(rsp, 0x20), rax);
+  __ movsd(xmm2, Address(rsp, 0x20));
+  __ comisd(xmm2, xmm0);
+  __ jcc(Assembler::below, L_1271);
+//   // y is NaN?
+//   if (!(b <= DP_CONST(7ff0000000000000))) return y + y;
+  __ mov64(rax, 0x7FF0000000000000);
+  __ movq(Address(rsp, 0x20), rax);
+  __ movsd(xmm2, Address(rsp, 0x20));
```

Comment:
You could directly do:
 __ movsd(xmm2, ExternalAddress((address)CONST_INF), rax);